lib/providers/shuttle_data.dart

```diff
@@ -3,49 +3,47 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/services.dart' show rootBundle;
 
 ///Observer Pattern
-///This class is used to parse the shuttle schedule JSON and get the next available shuttle bus.
+///This class is used to parse the shuttle schedule JSON and get the next available scheduled bus.
 class ShuttleData extends ChangeNotifier {
-  static Map schedule;
+  Map shuttleSchedule;
 
-  ShuttleData();
+  ShuttleData() {
+    // set schedule whenever rootBundle is done
+    rootBundle
+        .loadString('assets/shuttleSchedule.json')
+        .then((data) => shuttleSchedule = json.decode(data));
+  }
 
-  Future<String> getNextShuttle(campus, [DateTime time]) async {
-    schedule =
-        json.decode(await rootBundle.loadString('assets/shuttleSchedule.json'));
+  String getNextShuttle(campus, [DateTime time]) {
```

Comment:
What happens if I do ShuttleData().getNextShuttle("sgw") and the shuttleSchedule.json is large?
Since the constructor does not wait for the .loadString to complete, getNextShuttle() will try to read a null shuttleSchedule.